test/e2e/specs/editor/blocks/comments.spec.js

```diff
@@ -141,35 +144,210 @@ test.describe( 'Comments', () => {
 		).toBeHidden();
 	} );
 
-	test.afterEach( async ( { requestUtils, commentsBlockUtils } ) => {
-		// Ideally, we'd set options in afterAll. Unfortunately, these
-		// aren't exposed via the REST API, so we have to set them through the
-		// relevant wp-admin screen, which involves page utils; but those are
-		// prohibited from beforeAll.
-		await commentsBlockUtils.setOption(
-			'page_comments',
-			previousPageComments
+	test( 'A button allows the block to switch from legacy mode to editable mode', async ( {
+		admin,
+		editor,
+		page,
+	} ) => {
+		await admin.createNewPost();
+		await editor.insertBlock( {
+			name: 'core/comments',
+			attributes: { legacy: true, textColor: 'vivid-purple' },
+		} );
+
+		const block = page.locator( 'role=document[name="Block: Comments"i]' );
+		const warning = block.locator( '.block-editor-warning' );
+		const placeholder = block.locator(
+			'.wp-block-comments__legacy-placeholder'
 		);
-		await commentsBlockUtils.setOption(
-			'comments_per_page',
-			previousCommentsPerPage
+
+		await expect( block ).toHaveClass( /has-vivid-purple-color/ );
+		await expect( warning ).toBeVisible();
+		await expect( placeholder ).toBeVisible();
+
+		await page.click( 'role=button[name="Switch to editable mode"i]' );
+
+		const commentTemplate = block.locator(
+			'role=document[name="Block: Comment Template"i]'
+		);
+		await expect( block ).toHaveClass( /has-vivid-purple-color/ );
+		await expect( commentTemplate ).toBeVisible();
+		await expect( warning ).toBeHidden();
+		await expect( placeholder ).toBeHidden();
+	} );
+
+	test( 'The block version is rendered if the legacy attribute is false', async ( {
+		page,
+		admin,
+		editor,
+		requestUtils,
+	} ) => {
+		await admin.createNewPost();
+		await editor.insertBlock( {
+			name: 'core/comments',
+			attributes: { legacy: false },
+		} );
+		const postId = await editor.publishPost();
+
+		// Create a comments for that post.
+		await requestUtils.createComment( {
+			content: `This is an automated comment`,
```

Comment:
Nitpick, no need for backticks 😅 

```suggestion
			content: 'This is an automated comment',
```